component/component/main.jsonnet

```diff
@@ -16,29 +16,40 @@ local sync_options = {
 };
 
 // https://syn.tools/syn/explanations/commodore-components/secrets.html
-local secrets = [
-  if params.secrets[s] != null then
-    kube.Secret(s) {} + com.makeMergeable(params.secrets[s])
-  for s in std.objectFields(params.secrets)
-];
+local secrets = std.filter(function(it) it != null, [
+  if params.secrets[name] != null then
+    local secret = params.secrets[name];
+    assert std.objectHas(secret, 'metadata') : "missing `.metadata` in secret '%s'" % name;
+    assert std.get(secret.metadata, 'namespace', '') != '' : "`.metadata.namespace` in secret '%s' cannot be empty" % name;
+    kube.Secret(name) {} + com.makeMergeable(secret)
+  for name in std.objectFields(params.secrets)
+]);
 
-local additionalResources = [
-  if params.additionalResources[s] != null then
-    local res = params.additionalResources[s];
-    kube._Object(res.apiVersion, res.kind, s) + com.makeMergeable(res)
-  for s in std.objectFields(params.additionalResources)
-];
+local additionalResources = std.filter(function(it) it != null, [
+  if params.additionalResources[name] != null then
+    local res = params.additionalResources[name];
+    kube._Object(res.apiVersion, res.kind, name) + com.makeMergeable(res)
+  for name in std.objectFields(params.additionalResources)
+]);
 
-local composites = [
+local composites = std.filter(function(it) it != null, [
   if params.composites[name] != null then
     local res = params.composites[name];
     kube._Object('apiextensions.crossplane.io/v1', 'CompositeResourceDefinition', name) + sync_options + com.makeMergeable(res)
   for name in std.objectFields(params.composites)
-];
+]);
+
+local compositions = std.filter(function(it) it != null, [
+  if params.compositions[name] != null then
+    local composition = params.compositions[name];
+    kube._Object('apiextensions.crossplane.io/v1', 'Composition', name) + sync_options + com.makeMergeable(composition)
```

Comment:
You could export this logic to an external library how it was done for component-crossplane. Take a look [here](https://github.com/projectsyn/component-crossplane).

Reply:
As long as it's just one occurrence I don't see a need to put that into a function.
I also didn't want to introduce a hard dependency on component-crossplane's exported library just for 1 line of code :)

Reply:
We might use more complex functions later on for composition. For now it is ok to skip now.

Reply:
Yes, but that's not relevant yet for this PR.